sequencer/src/block.rs

```diff
@@ -1,144 +1,87 @@
-use crate::{NMTRoot, NamespaceProofType, Transaction, TransactionNMT, VmId, MAX_NMT_DEPTH};
+use crate::{BlockBuildingSnafu, Transaction};
 use commit::{Commitment, Committable};
-
 use hotshot_query_service::availability::QueryablePayload;
-use hotshot_types::{traits::block_contents::BlockPayload, utils::BuilderCommitment};
-use jf_primitives::merkle_tree::{namespaced_merkle_tree::NamespacedMerkleTreeScheme, prelude::*};
-use serde::{Deserialize, Deserializer, Serialize, Serializer};
-use std::fmt::{Debug, Display};
-
-#[derive(Clone, Debug, Deserialize, Serialize, Hash, PartialEq, Eq)]
-pub struct Payload {
-    #[serde(with = "nmt_serializer")]
-    pub(crate) transaction_nmt: TransactionNMT,
-}
-
-mod nmt_serializer {
-    use super::*;
-
-    // Serialize the NMT as a compact Vec<Transaction>
-    pub fn serialize<S>(nmt: &TransactionNMT, s: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        let leaves = nmt.leaves().cloned().collect::<Vec<Transaction>>();
-        leaves.serialize(s)
-    }
-
-    pub fn deserialize<'de, D>(deserializer: D) -> Result<TransactionNMT, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        use serde::de;
-
-        let leaves = <Vec<Transaction>>::deserialize(deserializer)?;
-        let nmt = TransactionNMT::from_elems(Some(MAX_NMT_DEPTH), leaves)
-            .map_err(|_| de::Error::custom("Failed to build NMT from serialized leaves"))?;
-        Ok(nmt)
-    }
-}
-
-impl QueryablePayload for Payload {
-    type TransactionIndex = u64;
-    type InclusionProof = <TransactionNMT as MerkleTreeScheme>::MembershipProof;
-    type Iter<'a> = Box<dyn Iterator<Item = u64>>;
-
-    fn len(&self, _meta: &Self::Metadata) -> usize {
-        self.transaction_nmt.num_leaves() as usize
-    }
-
-    fn transaction_with_proof(
-        &self,
-        _meta: &Self::Metadata,
-        index: &Self::TransactionIndex,
-    ) -> Option<(Self::Transaction, Self::InclusionProof)> {
-        match self.transaction_nmt.lookup(index) {
-            LookupResult::Ok(txn, proof) => Some((txn.clone(), proof)),
-            _ => None,
-        }
-    }
-
-    fn iter(&self, meta: &Self::Metadata) -> Self::Iter<'_> {
-        Box::new(0..self.len(meta) as u64)
-    }
-}
-
-impl BlockPayload for Payload {
-    type Error = bincode::Error;
+use hotshot_types::traits::BlockPayload;
+use hotshot_types::utils::BuilderCommitment;
+use serde::{Deserialize, Serialize};
+use snafu::OptionExt;
+
+pub mod entry;
+pub mod payload;
+pub mod queryable;
+pub mod tables;
+pub mod tx_iterator;
+
+use crate::block::entry::TxTableEntryWord;
+use crate::block::tables::NameSpaceTable;
```

Comment:
Nit: can just `use entry::...` or `use tables::....` since the modules are defined here

Reply:
I blame rust-analyzer for that. 😁